File: python/9_PalindromeNumber.py
def ref_isPalindrome():
    return 0

# Runtime: 64 ms, faster than 61.99% of Python3 online submissions for Palindrome Number.
# Memory Usage: 14.3 MB, less than 48.05% of Python3 online submissions for Palindrome Number.
def my_isPalindrome(x: int) -> bool:
    if x < 0:
        return False
    # read from median pos
    l = len(str(x))
    i = l/2 - 1 if l % 2 == 0 else (l-1)/2 -1
    # even: 10,8,6,4,2
    # odd :  9,7,5,3,1
    j = 0
    s = str(x)
    while(j<=i):
        if s[j] != s[l-j-1]:
            return False
        j+=1
    return True



if __name__ == "__main__":
    i = [0,12321]
    for I in i:
        print(isPalindrome(I))

File: cpp/9_PalindromeNumber.cpp
#include <bits/stdc++.h>
using namespace std;
// Runtime: 0 ms, faster than 100.00 % of C++ online submissions for Palindrome Number.
// Memory Usage : 5.9 MB, less than 74.53 % of C++ online submissions for Palindrome Number.
bool ref_isPalindrome(int x)
{
    if (x < 0) {
        return false;
    }
    
    int rev = 0;
    int temp = x;
    while (temp != 0) {
        int digit = temp % 10;
        temp /= 10;
        if (rev > 2147483647 / 10)
        {
            return 0;
        }
        if (rev < -2147483647 / 10)
        {
            return 0;
        }
        rev = rev * 10 + digit;
    }
    if (rev == x)
    {
        return true;
    }
    else
    {
        return false;
    }
    
}


// Runtime: 8 ms, faster than 79.24% of C++ online submissions for Palindrome Number.
// Memory Usage : 6.1 MB, less than 12.06 % of C++ online submissions for Palindrome Number.
bool my_isPalindrome(int x)
{
    if (x<0) 
        return 0;
    string s = to_string(x);
    int len = s.length();
    int l =  % 2 == 0 ? len/2 - 1 : len-1/2 -1
    int j = 0;
    while (j <= l)
    {
        if (s[j] != s[l - j - 1])
            return false;
        j++;
    }

    return true;             
}

int main(){
    
    return 0;
}
// Use this template always on leetcode to improve runtime.
static const int _ = []() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
   return 0;
}();